### PR TITLE
more compression types

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Add Xpress and ZSTD compression types.
@siddontang @ngaut PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@lishihai9017 PTAL

File: librocksdb_sys/src/lib.rs [hunk 1]
```diff
@@ -62,6 +62,8 @@ pub enum DBCompressionType {
     DBBz2 = 3,
     DBLz4 = 4,
     DBLz4hc = 5,
+    DBXpress = 6,
```
Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: do we support Xpress? maybe we should remove it here.

File: librocksdb_sys/build.sh [hunk 2]
```diff
@@ -105,6 +105,20 @@ function compile_lz4() {
...
+function compile_zstd() {
+    if [[ -f libzstd.a ]]; then
+        return
+    fi
+
+    rm -rf zstd-1.2.0
+    download https://github.com/facebook/zstd/archive/v1.2.0.tar.gz zstd-1.2.0.tar.gz d7777b0aafa7002a4dee1e2db42afe30
+    tar xvzf zstd-1.2.0.tar.gz
+    cd zstd-1.2.0
+    make CFLAGS='-fPIC' -j $con
```
Comment (siddontang) [inline librocksdb_sys/build.sh]: need -O2 or -O3 here?

Comment (zhangjinpeng1987) [inline librocksdb_sys/build.sh]: O3 by default.

**Comment by zhangjinpeng1987** [comment]
@siddontang PTAL

File: librocksdb_sys/src/lib.rs [hunk 3]
```diff
@@ -62,6 +62,8 @@ pub enum DBCompressionType {
     DBBz2 = 3,
     DBLz4 = 4,
     DBLz4hc = 5,
+    // DBXpress = 6, not support currently.
+    DBZSTD = 7,
```
Comment (andelf) [inline librocksdb_sys/src/lib.rs]: use consistent naming style.

DBZstd, since you've title-cased above all.

Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: +1

**Comment by siddontang** [comment]
Rest LGTM

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @andelf

**Comment by andelf** [comment]
LGTM

**Pull request merged**